pyresample/area_config.py

```diff
@@ -849,3 +850,37 @@ def convert_def_to_yaml(def_area_file, yaml_area_file):
     with open(yaml_area_file, 'w') as yaml_file:
         for area in areas:
             yaml_file.write(area.create_areas_def())
+
+
+def generate_area_def_rst_list(area_file):
+    """Create rst list of available area definitions with overview plot.
+
+    Args:
+        area_file (str): Path to area yaml file.
+
+    Returns:
+        str: rst list formatted string.
+    """
+    area_list = []
+
+    template = ("{area_name}\n"
+                "{n:^>{header_title_length}}\n\n"
+                ".. raw:: html\n\n"
+                "{content}\n\n"
+                "     <hr>\n\n")
+
+    for i, (aname, params) in enumerate(_read_yaml_area_file_content(area_file).items()):
+        area = _create_area_def_from_dict(aname, params)
+        if not hasattr(area, "_repr_html_"):
+            continue
+
+        if i == 0:
+            area_rep = area_repr(area, include_header=False, include_static_files=False)
+        else:
+            area_rep = area_repr(area, include_header=False)
```

Comment:
So static files are included for every row except the first? Or is this logic inverted from what it should be? Also, doesn't the `i == 0` check fail if we were given some swath definitions or other non-AreaDefinition (no `_repr_html_` method) geometries? I guess that's not technically possible right now (an area file with non-areas), but it could be in the future.

Reply:
Yes you are right about the static files logic. Totally missed that one. Fixed it.

About `i == 0` check: this could be a possible break point if non-areas could be possible in an area file in the future. This is based purely on what is in the area file right now. I don't really know what you have in mind for non-areas so I am not sure what's the smartest way to go here.

Reply:
I'm just thinking if we allowed users to make a SwathDefinition from YAML that loads binary lon/lat array files (an example off the top of my head) that this current logic could break. Given the logic is simple and saying "is this the first area we're printing", maybe the logic could be something not based on the index of the area in the list?